### ISSUE TITLE
panic during eval

### ISSUE THREAD

**Issue by ihard** [go-graphite/carbonapi#302]
carbonapi 0.10.0.1
config:
```
listen: "0.0.0.0:8081"
concurency: 200
cache:
   type: "mem"
   size_mb: 2048
   defaultTimeoutSec: 60
   memcachedServers:
       - "127.0.0.1:1234"
       - "127.0.0.2:1235"
cpus: 10
tz: ""
sendGlobsAsIs: true
maxBatchSize: 100000000
graphite:
    host: "graphite:2003"
    interval: "60s"
    prefix: "dc.host.carbon-api.info"
    pattern: "{prefix}"

idleConnections: 50
pidFile: ""
upstreams:
    buckets: 10

    timeouts:
        global: "30s"
        afterStarted: "20s"
        connect: "200ms"

    concurrencyLimit: 0
    keepAliveInterval: "30s"
    maxIdleConnsPerHost: 100

    backends:
        - "http://127.0.0.1:9090"

    carbonsearch:
        backend: ""
        prefix: ""
    graphite09compat: false
expireDelaySec: 300
logger:
    - logger: ""
      file: "/var/log/carbonapi/carbonapi.log"
      level: "info"
      encoding: "json"
```
some requests cause this error:
```
{"level":"ERROR","timestamp":"2018-03-19T10:17:32.111+0300","logger":"render","message":"panic during eval:","cache_key":"format=json&from=1520232368&maxDataPoints=1912&target=movingAverage%28consolidateBy%28groupByNode%28%2A.%2A.app1.processes.user.m1.m2.m3.%2A.%2A.count%2C+8%2C+%27sum%27%29%2C+%27sum%27%29%2C+%275min%27%29&until=1521443888","reason":"runtime error: index out of range","stack":"main.renderHandler.func3.1\n\t/root/go/src/github.com/go-graphite/c
arbonapi/http_handlers.go:408\nruntime.call32\n\t/usr/local/go/src/runtime/asm_amd64.s:509\nruntime.gopanic\n\t/usr/local/go/src/runtime/panic.go:491\nruntime.panicindex\n\t/usr/local/go/src/runtime/panic.go:28\ngithub.com/go-graphite/carbonapi/expr/functi
ons/moving.(*moving).Do\n\t/root/go/src/github.com/go-graphite/carbonapi/expr/functions/moving/function.go:67\ngithub.com/go-graphite/carbonapi/expr.EvalExpr\n\t/root/go/src/github.com/go-graphite/carbonapi/expr/expr.go:52\nmain.renderHandler.func3\n\t/roo
t/go/src/github.com/go-graphite/carbonapi/http_handlers.go:412\nmain.renderHandler\n\t/root/go/src/github.com/go-graphite/carbonapi/http_handlers.go:420\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:1918\nnet/http.(*ServeMux).Serv
eHTTP\n\t/usr/local/go/src/net/http/server.go:2254\ngithub.com/go-graphite/carbonapi/vendor/github.com/gorilla/handlers.CompressHandlerLevel.func1\n\t/root/go/src/github.com/go-graphite/carbonapi/vendor/github.com/gorilla/handlers/compress.go:146\nnet/http
.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:1918\ngithub.com/go-graphite/carbonapi/vendor/github.com/gorilla/handlers.(*cors).ServeHTTP\n\t/root/go/src/github.com/go-graphite/carbonapi/vendor/github.com/gorilla/handlers/cors.go:128\ngith
ub.com/go-graphite/carbonapi/vendor/github.com/gorilla/handlers.ProxyHeaders.func1\n\t/root/go/src/github.com/go-graphite/carbonapi/vendor/github.com/gorilla/handlers/proxy_headers.go:59\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.
go:1918\nnet/http.serverHandler.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2619\nnet/http.(*conn).serve\n\t/usr/local/go/src/net/http/server.go:1801"}
```

**Comment by Civil** [comment]
I think that's related to https://github.com/go-graphite/carbonapi/issues/215

currently carbonapi doesn't ensure that all the series have the same lenght (as graphite-web do) and as there is no check for that - it will panic.

There is a plan to fix that in near (hopefully) future, and there is an experimental flag `extrapolateExperiment` that will enable some functions (mostly ones that do aggregation) to behave the way graphite-web do, but it's untested and approach is not general.

**Comment by Civil** [comment]
Actually this issue should be fixed (or mostly fixed) by #496

Please open a new issue if this still happens with current master (or with release newer than 0.14.0).
